internal/services/datafactory/data_factory_linked_service_sql_managed_instance_resource_test.go

```diff
@@ -0,0 +1,360 @@
+	data := acceptance.BuildTestData(t, "azurerm_data_factory_linked_service_sql_managed_instance", "test")
+	r := LinkedServiceSQLManagedInstanceResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.servicePrincipal(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("service_principal_key"),
+	})
+}
+
+func TestAccDataFactoryLinkedServiceSQLManagedInstance_keyVaultPassword(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_factory_linked_service_sql_managed_instance", "test")
+	r := LinkedServiceSQLManagedInstanceResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.keyVaultPassword(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDataFactoryLinkedServiceSQLManagedInstance_keyVaultConnectionString(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_factory_linked_service_sql_managed_instance", "test")
+	r := LinkedServiceSQLManagedInstanceResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.keyVaultConnectionString(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("key_vault_password.0.linked_service_name", "key_vault_connection_string.0.secret_name"),
+	})
+}
+
+func (t LinkedServiceSQLManagedInstanceResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := linkedservices.ParseLinkedServiceID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.DataFactory.LinkedServiceClient.Get(ctx, id.ResourceGroupName, id.FactoryName, id.LinkedServiceName, "")
+	if err != nil {
+		return nil, fmt.Errorf("reading Data Factory SQL Managed Instance(%s): %+v", *id, err)
+	}
+
+	return pointer.To(resp.ID != nil), nil
+}
+
+func (LinkedServiceSQLManagedInstanceResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-df-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_data_factory" "test" {
+  name                = "acctestdf%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_data_factory_linked_service_sql_managed_instance" "test" {
+  name              = "acctestlssqlmi%[1]d"
+  data_factory_id   = azurerm_data_factory.test.id
+  connection_string = "Server=myserver.database.windows.net;Database=mydatabase;User ID=myuser;Password=mypassword"
+  annotations       = ["test1", "test2", "test3"]
+  description       = "test description"
+
+  parameters = {
+    param1 = "value1"
+    param2 = "value2"
+  }
```

Comment:
since this is the basic test, we should remove these and include only required properties